libkbfs/root_metadata.go

```diff
@@ -694,7 +699,15 @@ func (md *RootMetadata) fillInDevices(crypto Crypto,
 
 func (md *RootMetadata) finalizeRekey(
 	crypto cryptoPure, prevKey, currKey kbfscrypto.TLFCryptKey) error {
-	return md.bareMd.FinalizeRekey(crypto, prevKey, currKey, md.extra)
+	wkbID := md.bareMd.GetTLFWriterKeyBundleID()
+	rkbID := md.bareMd.GetTLFReaderKeyBundleID()
+	err := md.bareMd.FinalizeRekey(crypto, prevKey, currKey, md.extra)
+	if err == nil && md.extra != nil {
+		includeWkb := wkbID != md.bareMd.GetTLFWriterKeyBundleID()
+		includeRkb := rkbID != md.bareMd.GetTLFReaderKeyBundleID()
+		md.extraNew = md.extra.Copy(includeWkb, includeRkb)
```

Comment:
It's a simple filter for which key bundles are new and need uploading so setting the pointers is fine.